### DISCUSSION TITLE
SX1261/2 setTCXO delay choice

### DISCUSSION THREAD

**Discussion by S5NC** [jgromes/RadioLib#1055] [General]
Why was 5000 us [chosen](https://github.com/jgromes/RadioLib/blob/263f7883cf222f8ccb79788e68a653eab17d88e6/src/modules/SX126x/SX126x.h#L898) as the default TCXO delay/timeout? Did lower values not work? The data sheets shows 500 us in the diagram when talking about it, but there is no mention of a recommend value.

![image](https://github.com/jgromes/RadioLib/assets/145265251/25c60248-67ad-4d7c-962c-1cdd279aeae4)

Were you getting `XOSC_START_ERR` at lower values or is this just an overcautious duration to ensure it works for all boards?

**Comment by jgromes** [comment]
I don't quite recall where the default value came from. It could well be overly cautious, but at the same time 5ms wait is not that much of a penalty.

On a side note, I don't think the note in the image implies that the 500us is the TCXO delay, in fact I would argue it's trying to tell you that the TCXO delay will be added to those 500us, which is just the time required to switch from Rx to sleep mode. Figure 13-2 shows the same 500 us without any mention of TCXO.

**Comment by HeadBoffin** [comment]
@S5NC, rather than opening up by interrogating people about historic decisions that are lost in the dense fog of technical options, why not do some testing of what works and once you have a concrete proposal - the why & the how, come back to add to this discussion.

At the absolute minimum, tell us what the 'why' is because otherwise it's just idle banter which most can ill afford to participate in.
